### PR TITLE
feat: wrap Melange Js / Belt / Node libraries

### PR DESCRIPTION
**Pull request by anmonteiro**
- after this change, Melange exposes fewer toplevel modules. Melange runtime / stdlib modules are now wrapped under:
    - `Caml*` / `Curry` modules are part of the runtime and keep being exposed as before
    - `Js.*` contains all the modules previously accessible via `Js_*`, e.g. `Js_int` -> `Js.Int`
    - `Belt.*` wraps all the `Belt` modules; `Belt_List` etc. are not exposed anymore, but rather nested under `Belt`, e.g. `Belt.List`
    - `Node.*`: we now ship a `melange.node` library that includes the modules containing Node.js bindings. After this change, users will have to depend on `melange.node` explicitly in order to use the `Node.*` modules
    - `Dom.*`: we now ship a `melange.dom` library that includes the modules containing Node.js bindings. This library is included by default so the `Dom` module will always be available in Melange projects.

fixes https://github.com/melange-re/melange/issues/566

### CODE REVIEW

**Comment by anmonteiro** [comment]
fixes https://github.com/melange-re/melange/issues/566

File: jscomp/others/node_process.ml [hunk 1]
```diff
@@ -32,7 +32,7 @@ type t =
     cwd : unit -> string [@bs.meth];
     disconnect : unit -> unit [@bs.meth];
     platform : string;
-    env : string Js.Dict.t; (* ocamldep sucks which can not map [Js.Dic.t] to [Js.Dict.t]*)
```
Comment (davesnx) [inline jscomp/others/node_process.ml]: lol

**Comment by psb** [comment]
Is this a typo?
> Dom.*: we now ship a melange.dom library that includes the modules containing Node.js bindings.

I.e., it's not nodejs bindings.

**Comment by anmonteiro** [comment]
Indeed a typo. Will fix the release notes.

**Pull request merged**